core/src/test/java/org/elasticsearch/common/io/stream/StreamInputTests.java

```diff
@@ -0,0 +1,213 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.common.io.stream;
+
+import org.elasticsearch.common.io.stream.StreamInput.StreamInputFunction;
+import org.elasticsearch.test.ESTestCase;
+
+import org.mockito.InOrder;
+import org.mockito.Matchers;
+
+import java.io.IOException;
+import java.util.function.IntFunction;
+import java.util.function.Supplier;
+
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+/**
+ * Tests {@link StreamInput}.
+ */
+public class StreamInputTests extends ESTestCase {
```

Comment:
Mockito isn't really required for testing these the way we do in BytesStreamsTests. which is always a plus.